README.md

```diff
@@ -1 +1,183 @@
+
+For example: 
+```js
+cy.tget('app_info:visible create_new_user_section submit_button');
+```
+will be converted automatically into 
+```js
+cy.get('[data-test=app_info]:visible [data-test=create_new_user_section] [data-test=submit_button]');
+```
+
+## Installation
+```bash
+pnpm add -D cypress-selector-shorthand
+# or
+npm i -D cypress-selector-shorthandgenerator
+# or
+yarn add -D cypress-selector-shorthand
+```
+
+Add this line to your project's `cypress/support/commands.js` (or `commands.ts`):
+```js
+import 'cypress-selector-shorthand/install';
+```
+
+## Configuration
+To make use of the navigation shorthands, there's three extra steps you need to do to generate the objects and interfaces. (Note: you can use `cy.tget()` without this.)
+
+First, you need to create a schema JSON file, e.g. the following might be used for the example at the top of this file:
+
+```json
+{
+    "app_info": {
+        "create_new_user_section": {
+            "name_input": null,
+            "address_input": null,
+            "submit_button": null
+        },
+        "existing_users_table": {
+            "rows" : {
+                "row": {
+                    "firstName": null,
+                    "lastName": null
+                }
+            }
+        }
+    }
+}
+```
+
+Once you have your schema, you need to generate interfaces for IntelliSense:
+
+```bash
+npx cypress-selector-shorthand interfaces --appName 'MyApp' --infile './schema.json' --outfile './src/schema.ts`
+```
+
+This will create a bunch of interfaces from your schema file (using the `quicktype` library):
+
+```typescript
+export interface MyApp {
+    appInfo: Cypress.ChainableLike<JQuery<HTMLElement>, AppInfo>;
+}
+
+export interface AppInfo {
+    createNewUserSection: Cypress.ChainableLike<JQuery<HTMLElement>, CreateNewUserSection>;
+    existingUsersTable:   Cypress.ChainableLike<JQuery<HTMLElement>, ExistingUsersTable>;
+}
+
+export interface CreateNewUserSection {
+    nameInput:    Cypress.Chainable<JQuery<HTMLElement>>
+    addressInput: Cypress.Chainable<JQuery<HTMLElement>>
+    submitButton: Cypress.Chainable<JQuery<HTMLElement>>
+}
+
+export interface ExistingUsersTable {
+    rows: Cypress.ChainableLike<JQuery<HTMLElement>, Rows>;
+}
+
+export interface Rows {
+    row:  Cypress.ChainableLike<JQuery<HTMLElement>, Row>;
+    with: With;
+}
+
+export interface Row {
+    firstName: Cypress.Chainable<JQuery<HTMLElement>>
+    lastName:  Cypress.Chainable<JQuery<HTMLElement>>
+}
+
+export interface With {
+    firstName: (text: string) => Cypress.ChainableLike<JQuery<HTMLElement>, Row>;
+    lastName:  (text: string) => Cypress.ChainableLike<JQuery<HTMLElement>, Row>;
+}
+```
+
+You might notice this generated a `with` field that didn't exist in the schema. More on that under the Usage section.
+
+Finally, you need to generate your navigation object:
```

Comment:
where should this code be placed in order for users to take advantage of the navigation object?

Reply:
location doesn't matter, but you need to import it. To reply to several of your questions at once:
If you run
`npx cypress-selector-shorthand interfaces --appName 'MyApp' --infile './myAppSchema.json' --outfile './src/myAppInterface.ts'`

then your schema file must be in `./myAppSchema.json` relative to your project directory, and you have to import `./src/myAppInterface.ts` when you create your `./src/myApp.ts` file.

When I update the PR, I have revised the README and hopefully that will provide some clarity